scripts/degradation-tester/main.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+var opThresholds map[string]float64
+var allocThresholds map[string]float64
+
+type Config struct {
+	DefaultOpDelta    float64                  `yaml:"DefaultOpDelta"`
+	DefaultAllocDelta float64                  `yaml:"DefaultAllocDelta"`
+	Packages          []BenchmarkingTestConfig `yaml:"Packages"`
+}
+
+type BenchmarkingTestConfig struct {
+	Path  string     `yaml:"Path"`
+	Tests []TestCase `yaml:"Tests"`
+}
+
+type TestCase struct {
+	Name       string  `yaml:"Name"`
+	OpDelta    float64 `yaml:"OpDelta"`
+	AllocDelta float64 `yaml:"AllocDelta"`
+}
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Println("Usage: degradation-tester <config_filename> <results_filename>")
+		os.Exit(1)
+	}
+
+	configFilename := os.Args[1]
+	resultsFilename := os.Args[2]
+
+	config := loadConfig(configFilename)
+
+	file, err := os.Open(resultsFilename)
+	if err != nil {
+		fmt.Printf("Error opening results file: %v\n", err)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	var currentSection string
+
+	scanner := bufio.NewScanner(file)
```

Comment:
try to run benchstat with CSV output format, instead of parsing lines
this could help reduce regex usage

https://cs.opensource.google/go/x/perf/+/5eefbfdb:cmd/benchstat/main.go;l=455